chirp_client.cc

```diff
@@ -0,0 +1,302 @@
+#include "chirp_client.h"
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <grpcpp/grpcpp.h>
+#include <google/protobuf/util/time_util.h>
+
+#include "service.grpc.pb.h"
+
+DEFINE_string(register, "", "register user in key value store");
+DEFINE_string(user, "", "current user of chirp");
+DEFINE_string(chirp, "", "string to chirp out");
+DEFINE_string(reply, "", "string to reply to a given chirp");
+DEFINE_string(read, "", "Reads the chirp thread starting at the given id");
+DEFINE_string(follow, "", "Starts following the given username");
+DEFINE_bool(monitor, false, "Streams new tweets from those currently followed");
+
+void ChirpClient::registeruser(const std::string& user) {
+  grpc::ClientContext context;
+  chirp::RegisterRequest request;
+  chirp::RegisterReply reply;   // Data from server will be updated here
+
+  request.set_username(user);
+  
+  // RPC call
+  grpc::Status status = stub_->registeruser(&context, request, &reply);
+
+  // User already exists in key value store
+  if(status.error_code() == grpc::StatusCode::ALREADY_EXISTS) {
+    std::cout << "Failure to register " << "'" << user << "'" << std::endl;
+    std::cout << status.error_message() << std::endl;
+    return;
+  }
+  // Registration was successful
+  if (status.ok()) {
+    std::cout << "'" << user << "' successfully registered" << std::endl;
+    return;
+  } 
+  else {
+    std::cout << "Failure to register " << "'" << user << "'" << std::endl;
+    std::cout << status.error_message() << std::endl;    
+    return;
+  }
+}
+
+void ChirpClient::chirp(const std::string& user, const std::string& text, const std::string& parent_id) {
+  grpc::ClientContext context;
+  chirp::ChirpRequest request;
+  chirp::ChirpReply *reply = new chirp::ChirpReply(); // Data from server will be updated here
+
+  request.set_username(user);
+  request.set_text(text);
+  request.set_parent_id(parent_id);
+
+  // RPC call
+  grpc::Status status = stub_->chirp(&context, request, reply);
```

Comment:
Same question about passing request pointer to server.

Reply:
I checked a few other peoples repos and I don't see them doing it. not 100% sure though